torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -1,72 +1,68 @@
-from typing import Tuple, List, Optional
+from typing import List, TypeVar, Any
 
 import torch
-from torchvision.transforms import (  # noqa: F401
-    functional as _F,
-    InterpolationMode,
-)
+from torchvision.prototype import features
+from torchvision.prototype.transforms import kernels as K
+from torchvision.transforms import functional as _F
+
+from ._utils import dispatch
 
-horizontal_flip_image = _F.hflip
+T = TypeVar("T", bound=features.Feature)
 
 
-def horizontal_flip_bounding_box(bounding_box: torch.Tensor, *, image_size: Tuple[int, int]) -> torch.Tensor:
-    shape = bounding_box.shape
-    bounding_box = bounding_box.view(-1, 4)
-    bounding_box[:, [0, 2]] = image_size[1] - bounding_box[:, [0, 2]]
-    return bounding_box.view(shape)
+def _horizontal_flip_bounding_box(input: features.BoundingBox) -> torch.Tensor:
```

Comment:
@pmeier why this one returns torch.Tensor and in `_resize_bounding_box` output is features.BoundingBox and wrapping is explicit.

Reply:
I think that might be a bug actually. IT should return a BoundingBox not a Tensor.